osu.Game/Screens/Select/Carousel/CarouselBeatmapSet.cs

```diff
@@ -4,30 +4,56 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using osu.Framework.Bindables;
 using osu.Framework.Extensions.IEnumerableExtensions;
 using osu.Game.Beatmaps;
 using osu.Game.Screens.Select.Filter;
+using osu.Game.Users;
 
 namespace osu.Game.Screens.Select.Carousel
 {
     public class CarouselBeatmapSet : CarouselGroupEagerSelect
     {
+        private readonly Bindable<User> localUser;
```

Comment:
As above - this ideally should be avoided / encapsulated somewhere IMO.

Also I fail to see the point of injecting this manually via ctor from `BeatmapCarousel` when you could use DI directly here & resolve.

Reply:
Not sure how DI works for classes that aren't drawables.

Reply:
It doesn't, which I suppose explains it. It's still really strange either way.